### ISSUE TITLE
Add support for fast image rendering on any linear axes

### ISSUE THREAD

**Issue by antoinerg** [plotly/plotly.js#5305]
At the moment, `image` trace will only get fast rendering if axes are compatible as per:
https://github.com/plotly/plotly.js/blob/f7baf5d2f3f2e960fca95460a2a119977db8c6f0/src/traces/image/plot.js#L34

However, `compatibleAxis` here is a bit too strict since there's no reason we couldn't support both reversed and normal linear axes:
https://github.com/plotly/plotly.js/blob/f7baf5d2f3f2e960fca95460a2a119977db8c6f0/src/traces/image/plot.js#L18-L22

Basically, we should :hocho: line 21 and then proceed to get the positioning right as well as the correct hover labels for any order of the axes :smile_cat:
